backend/src/services/auth/auth.service.ts

```diff
@@ -0,0 +1,13 @@
+import { IUser } from 'healthcare-shared/common/interfaces/user';
+import { userRepository } from '~/data/repositories';
+import { SignUpResponse } from 'healthcare-shared/common/types'
+
+class AuthService {
+  public async signUp({ userCandidate }: { userCandidate: IUser }): Promise<SignUpResponse> {
```

Comment:
```suggestion
  public async signUp(registerPayload: IRegisterPayload): Promise<SignUpResponse> {
```
You also need to incode password